### PR TITLE
drainer: Add a metric to track the delay of the downstream

### PR DESCRIPTION
**Pull request by suzaku**
### What problem does this PR solve? 

Fix [Jira Issue](https://internal.pingcap.net/jira/browse/TOOL-1170)

### What is changed and how it works?

A new metric `checkpoint_delay = now() - most-recent-ts` is included.

### Check List 

Tests 

Manual test (confirmed that the new metric appeared in the /metrics endpoint):  
```
# HELP binlog_drainer_checkpoint_delay_seconds How much the downstream checkpoint lag behind
# TYPE binlog_drainer_checkpoint_delay_seconds gauge
binlog_drainer_checkpoint_delay_seconds 15.128
```

Code changes

Side effects

Related changes

### CODE REVIEW

File: drainer/metrics.go [hunk 1]
```diff
@@ -74,6 +74,14 @@ var (
 			Help:      "save checkpoint tso of drainer.",
 		})
 
+	checkpointDelayGauge = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: "binlog",
+			Subsystem: "drainer",
+			Name:      "checkpoint_delay",
```
Comment (july2993) [inline drainer/metrics.go]: better let people know the unit, like name `check_delay_seconds`

Comment (suzaku) [reply drainer/metrics.go]: Fixed

**Comment by WangXiangUSTC** [comment]
LGTM

**Comment by suzaku** [comment]
> please update https://github.com/pingcap/tidb-ansible/blob/master/scripts/binlog.json
> and test it.
> you can import the binlog.json, edit it and export it.

I'll update the Grafana dashboard later.
It should be OK to merge this PR now, because it's mainly aimed to be used directly by querying the metric interface.

File: drainer/metrics.go [hunk 2]
```diff
@@ -69,6 +69,15 @@ var (
 			Help:      "save checkpoint tso of drainer.",
 		})
 
+	checkpointDelayHistogram = prometheus.NewHistogram(
+		prometheus.HistogramOpts{
+			Namespace: "binlog",
+			Subsystem: "drainer",
+			Name:      "checkpoint_delay_seconds",
+			Help:      "How much the downstream checkpoint lag behind",
+			Buckets:   prometheus.ExponentialBuckets(0.00005, 2, 18),
```
Comment (july2993) [inline drainer/metrics.go]: use `prometheus.ExponentialBuckets(0.001, 2, 22), // 1ms ~ 1hours`

**Pull request merged**